src/Compilers/CSharp/Test/Emit/PrivateProtected.cs

```diff
@@ -452,5 +576,218 @@ private protected interface I {}                       // 8
                 .VerifyDiagnostics(
                 );
         }
+
+        [Fact]
+        public void VerifyPrivateProtectedIL()
+        {
+            var text = @"
+class Program
+{
+    private protected void M() {}
+    private protected int F;
+}
+";
+            var verifier = CompileAndVerify(
+                text,
+                parseOptions: TestOptions.Regular7_2,
+                expectedSignatures: new[]
+                {
+                    Signature("Program", "M", ".method famandassem hidebysig instance System.Void M() cil managed"),
+                    Signature("Program", "F", ".field famandassem instance System.Int32 F"),
+                });
+        }
+
+        [Fact]
+        public void VerifyPartialPartsMatch()
+        {
+            var source =
+@"class Outer
+{
+    private protected partial class Inner {}
+    private           partial class Inner {}
+}";
+            CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2)
+                .VerifyDiagnostics(
+                // (3,37): error CS0262: Partial declarations of 'Outer.Inner' have conflicting accessibility modifiers
+                //     private protected partial class Inner {}
+                Diagnostic(ErrorCode.ERR_PartialModifierConflict, "Inner").WithArguments("Outer.Inner").WithLocation(3, 37)
+                );
+            source =
+@"class Outer
+{
+    private protected partial class Inner {}
+    private protected partial class Inner {}
+}";
+            CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2)
+                .VerifyDiagnostics(
+                );
+        }
+
+        [Fact]
+        public void VerifyProtectedSemantics()
+        {
+            var source =
+@"class Base
+{
+    private protected void M()
+    {
+        System.Console.WriteLine(this.GetType().Name);
+    }
+}
+
+class Derived : Base
+{
+    public void Main()
+    {
+        Derived derived = new Derived();
+        derived.M();
+        Base bb = new Base();
+        bb.M(); // error 1
+        Other other = new Other();
+        other.M(); // error 2
+    }
+}
+
+class Other : Base
+{
+}";
+            CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2)
+                .VerifyDiagnostics(
+                // (16,12): error CS1540: Cannot access protected member 'Base.M()' via a qualifier of type 'Base'; the qualifier must be of type 'Derived' (or derived from it)
```

Comment:
>qualifier [](start = 90, length = 9)

I understand this is an existing message, but the "qualifier" term didn't make any sense to me. What does it mean?
Consider filing an issue to improve the message if you think that's appropriate.

Reply:
The qualifier is the left-hand-side of a qualified name.

---
In reply to: [137604439](https://github.com/dotnet/roslyn/pull/21742#discussion_r137604439) [](ancestors = 137604439)

Reply:
Opened #21965 to track this.